### PR TITLE
Add high-level API wrapper: Circdata

### PR DESCRIPTION
**Pull request by HeinrichHartmann**
The Classes in the circdata module allow convenient access to the most common data fetching operations:
- Search for metrics + fetch data via the `/metrics` and `/data` API endpoints
- CAQL via `/caql`

The output of this methods can directly be passed to pandas for further analysis.
Histogram output is supported. A blog post/examples on how to appy this functions is forthcoming.

An effort is made to treat histogram and numeric values on equal footing. When a search results sets are hybrid numeric/histogram we convert the output to the requested kind.

The circdata module depends on circllhist to be available on the system.
At the time of this writing circllhist can't be installed via pip, so we stay away from making it a dependency in setup.py.

I am not sure how to best deal with dependencies of optional additions like circdata.py.
Even without circllhist, the CirconusAPI class is fully usable.

### CODE REVIEW

File: circonusapi/circonusdata.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+Circonus Data Fetching API
+
+This module provides simplified methods for data fetching operations.
+To gain access to the full functionality use the circonusapi module.
+"""
+
+import math
+from datetime import datetime
+
+from circonusapi import circonusapi
```
Comment (cody271) [inline circonusapi/circonusdata.py]: Need to explicitly use a relative import here for Python 2 and 3 compatibility.
```diff
iff --git a/circonusapi/circonusdata.py b/circonusapi/circonusdata.py
index 58e1eb5..9839023 100644
--- a/circonusapi/circonusdata.py
+++ b/circonusapi/circonusdata.py
@@ -8,7 +8,7 @@ To gain access to the full functionality use the circonusapi module.
 import math
 from datetime import datetime
 
-from circonusapi import circonusapi
+from . import circonusapi
 from circllhist import Circllhist
 ```

File: circonusapi/circonusdata.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+from circllhist import Circllhist
```
Comment (cody271) [inline circonusapi/circonusdata.py]: Probably need to add `circllhist` to `setup.py`, since it is not an optional dependency.

**Comment by cody271** [comment]
Also we should probably pull the two main classes forward into `__init__.py`:
```diff
diff --git a/circonusapi/__init__.py b/circonusapi/__init__.py
index e69de29..1080f72 100644
--- a/circonusapi/__init__.py
+++ b/circonusapi/__init__.py
@@ -0,0 +1,3 @@
+from .circonusapi import CirconusAPI
+from .circonusdata import CirconusData
+__all__ = ["CirconusAPI", "CirconusData"]
```

File: circonusapi/circonusdata.py (hunk 2 above)
Comment (HeinrichHartmann) [reply circonusapi/circonusdata.py]: I would like to keep the dependency optional. `circonusapi.py` if fully functional without circllhist. Only circounsdata is affected.

**Comment by HeinrichHartmann** [comment]
We can't pull the imports into `__init__` if we want to have the circllhist dependency optional.

**Pull request merged**